internal/server/v2/observation/calculation_util.go

```diff
@@ -0,0 +1,46 @@
+// Copyright 2024 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package observation
+
+import (
+	pbv2 "github.com/datacommonsorg/mixer/internal/proto/v2"
+)
+
+// Given an input ObservationResponse, generate a map of variable -> entities with missing data.
+func findObservationResponseHoles(inputReq *pbv2.ObservationRequest, inputResp *pbv2.ObservationResponse) map[string]*pbv2.DcidOrExpression {
+	result := map[string]*pbv2.DcidOrExpression{}
+	if inputReq.Variable.GetFormula() != "" {
+		// Currently do not support nested formulas.
```

Comment:
shall we raise exception here to make this more explicit?

Reply:
Done